contracts/ToDoManager.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity ^0.5.0;
+
+contract ToDoManager {
+    
+    enum Status {initial, toDo, inProgress, done}
+    mapping (uint => string) indexToDo;
+    mapping (string => Status) toDoStatus;
+    uint public indexCounter;
+    mapping (string => address) assignedToDo; 
+    
+    modifier onlyAssignee(uint _index) {
+        string memory _toDo = indexToDo[_index];
+        require(msg.sender == assignedToDo[_toDo]);
+        _;
+    }
+    
+    function addToDo(string memory _toDo) public {
```

Comment:
I think id-s of the to-dos should be just their indexes and you should be able to add one and the same todo twice like "buy groceries". Strings in general are not great thing to be a key in mapping

Reply:
I think it would be best to get the todo topic, and return an the index as identificator. For the other methods request the id to be passed